src/main/scala/scalismo/mesh/boundingSpheres/SurfaceSpatialIndex.scala

```diff
@@ -111,9 +112,9 @@ private case class ClosestPointMeta(val distance2: Double,
 /**
  * Surface distance implementation for TriangleMesh3D.
  */
-private case class TriangleMesh3DSpatialIndex(bs: BoundingSphere,
+private[mesh] case class TriangleMesh3DSpatialIndex(bs: BoundingSphere,
```

Comment:
I suggest to remove case here: the class does not support the "value"-type equality and hash semantics one would expect due to the BoundingSphere member (usage in TriangleMesh3DOperations is already with new anyway)